homeassistant/components/insteon/climate.py

```diff
@@ -0,0 +1,232 @@
+    """A Class for an Insteon climate entity."""
+
+    @property
+    def supported_features(self):
+        """Return the supported features for this entity."""
+        return SUPPORTED_FEATURES
+
+    @property
+    def temperature_unit(self) -> str:
+        """Return the unit of measurement."""
+        if self._insteon_device.properties[CELSIUS].value:
+            return TEMP_CELSIUS
+        return TEMP_FAHRENHEIT
+
+    @property
+    def current_humidity(self) -> Optional[int]:
+        """Return the current humidity."""
+        return self._insteon_device.groups[HUMIDITY].value
+
+    @property
+    def hvac_mode(self) -> str:
+        """Return hvac operation ie. heat, cool mode."""
+        return HVAC_MODES[self._insteon_device.groups[SYSTEM_MODE].value]
+
+    @property
+    def hvac_modes(self) -> List[str]:
+        """Return the list of available hvac operation modes."""
+        return list(HVAC_MODES.values())
+
+    @property
+    def current_temperature(self) -> Optional[float]:
+        """Return the current temperature."""
+        return self._insteon_device.groups[TEMPERATURE].value
+
+    @property
+    def target_temperature(self) -> Optional[float]:
+        """Return the temperature we try to reach."""
+        if self._insteon_device.groups[SYSTEM_MODE].value == ThermostatMode.HEAT:
+            return self._insteon_device.groups[HEAT_SET_POINT].value
+        if self._insteon_device.groups[SYSTEM_MODE].value == ThermostatMode.COOL:
+            return self._insteon_device.groups[COOL_SET_POINT].value
+        return None
+
+    @property
+    def target_temperature_high(self) -> Optional[float]:
+        """Return the highbound target temperature we try to reach."""
+        if self._insteon_device.groups[SYSTEM_MODE].value == ThermostatMode.AUTO:
+            return self._insteon_device.groups[COOL_SET_POINT].value
+        return None
+
+    @property
+    def target_temperature_low(self) -> Optional[float]:
+        """Return the lowbound target temperature we try to reach."""
+        if self._insteon_device.groups[SYSTEM_MODE].value == ThermostatMode.AUTO:
+            return self._insteon_device.groups[HEAT_SET_POINT].value
+        return None
+
+    @property
+    def fan_mode(self) -> Optional[str]:
+        """Return the fan setting."""
+        return FAN_MODES[self._insteon_device.groups[FAN_MODE].value]
+
+    @property
+    def fan_modes(self) -> Optional[List[str]]:
+        """Return the list of available fan modes."""
+        return list(FAN_MODES.values())
+
+    @property
+    def target_humidity(self) -> Optional[int]:
+        """Return the humidity we try to reach."""
+        high = self._insteon_device.groups[HUMIDITY_HIGH].value
+        low = self._insteon_device.groups[HUMIDITY_LOW].value
+        # May not be loaded yet so return a default if required
+        return (high + low) / 2 if high and low else 50
+
+    @property
+    def min_humidity(self) -> int:
```

Comment:
If you are implementing `min_humidity`, do you need `max_humidity`?

Reply:
Yes because this device does not have a true target humidity as a single number. It has a low and a high. If it goes below the low then it triggers a humidifier. If it goes above the high then it triggers a dehumidifier.

Reply:
Oh, now I see your issue. Yes but the default `max_humidity` is 99 which is the `max_humidity` for this device so I don't need to override the `max_humidity` in the `ClimateEntity` class.